gsy_framework/read_user_profile.py

```diff
@@ -332,13 +332,14 @@ def read_arbitrary_profile(profile_type: InputProfileTypes,
             GlobalConfig.IS_CANARY_NETWORK:
         profile = _copy_profile_to_multiple_days(profile)
     if profile is not None:
-        if profile_type in [InputProfileTypes.POWER_W, InputProfileTypes.REBASE_W]:
+        if profile_type is not InputProfileTypes.ENERGY_KWH:
```

Comment:
```suggestion
        if profile_type not in [InputProfileTypes.ENERGY_KWH, InputProfileTypes.IDENTITY]:
```
Unless I misunderstood here and you would prefer to fill identity profiles too. In this case though, why not include the energy profiles from `_fill_gaps_in_profile` ?

Reply:
> Unless I misunderstood here and you would prefer to fill identity profiles too

Yes, this was my goal in order to be backwards compatible

> In this case though, why not include the energy profiles to _fill_gaps_in_profile

We discussed this and said that this is implicit behaviour and should not be done really. For the "new" energy profiles I thought we should be correct from the start.
Or do you think that we validate the gsy-web profiles anyway and there will not be any gaps from that side anyway?

Reply:
Ok , now I recall our discussion. The problem I have with this is that we treat energy and identity profiles differently, which might make debugging more difficult in the long run (one would have to remember that gap-filling only applies to identity and not energy profiles). I agree with you that the only problem are profiles that already exist on the database, for all new profiles we should not have this problem as far as I recall. Anyways, because this discussion is probably an overkill, it is fine to stay as is, we will probably anyways remove this from identity profiles later down the road.